Selective invoice finance loan writeback (#824)
codatio/codat-docs: docs/lending/guides/loan-writeback/repay-invoice-finance.md

---
title: "Record invoice finance repayments"
description: "Record the repayment of money owed to the lender for an invoice financing loan in the SMB's accounting platform"
sidebar_label: "Record: invoice finance"
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import WritebackMapping from "@components/global/Prototypes/WritebackMapping";

Once the borrower's customer has paid for the goods or services they purchased, the financed invoice is ready to be reconciled in the accounting platform.  

To reflect that programmatically, perform these steps:

> 1. [Create a transfer](/lending/guides/loan-writeback/repay-inoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.

Author: ```suggestion
1. [Create a transfer](/lending/guides/loan-writeback/repay-invoice-finance#create-transfer) from the lender's bank account to the borrower's to account for the oustanding amount less fees.
```

> 2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-inoice-finance#create-direct-cost) against the lender's bank account.

Author: ```suggestion
2. To record interest or fees, [create a direct cost](/lending/guides/loan-writeback/repay-invoice-finance#create-direct-cost) against the lender's bank account.
```

> 3. To update the invoice as paid, [create a payment](/lending/guides/loan-writeback/repay-inoice-finance#create-payment) in the lender's bank account.

Author: ```suggestion
3. To update the invoice as paid, [create a payment](/lending/guides/loan-writeback/repay-invoice-finance#create-payment) in the lender's bank account.
```

> 4. [Create bank transactions](/lending/guides/loan-writeback/repay-inoice-finance#create-bank-transactions) to keep the lender's bank account transactions in sync.

Author: ```suggestion
4. [Create bank feed transactions](/lending/guides/loan-writeback/repay-invoice-finance#create-bank-feed-transactions) to represent the transfer, direct cost and invoice payment in the lender's bank account.
```

```mermaid
sequenceDiagram
    participant backend as Your application 
    participant codat as Codat
    
        backend ->> codat: Create transfer (lender -> bank account)
        codat -->> backend:  transfer

        backend ->> codat: Create direct cost
        codat -->> backend: direct cost
        
        backend ->> codat: Create payment (on invoice)
        codat -->> backend: payment
        

>         backend ->> codat: Create bank transactions (transfer, direct cost and invoice payment amounts)

Author: ```suggestion
        backend ->> codat: Create bank feed transactions (transfer, direct cost and invoice payment amounts)
```

>         codat -->> backend: bank transactions

Author: ```suggestion
        codat -->> backend: bank feed transactions
```

                        Type: shared.PaymentLinkTypeInvoice,
                    },
                },
            },
        },
        TotalAmount: types.MustNewDecimalFromString(invoice.TotalAmount),
    },
    CompanyID: "8a210b68-6988-11ed-a1eb-0242ac120002",
    ConnectionID: "2e9d2c44-f675-40ba-8049-353bfcb5e171",
})
```
</TabItem>

<TabItem value="http" label="HTTP">

```http
POST https://api.codat.io/companies/{companyId}/connections/{connectionId}/push/payments
```

#### Request body

```json
{
  "customerRef": {
    "id": invoice.customerRef.id,
  },
  "accountRef": {
    "id": lendersBankAccount.id,
  },
  "totalAmount": invoice.totalAmount,
  "currency": borrowersBankAccount.currency,
  "currencyRate": 1,
  "date": invoicePaymentDate,
  "lines": [
    {
      "amount": invoice.totalAmount, // The sum of amount and line amounts must equal zero
      "links": [
        {
          "type": "Invoice",
          "id": invoice.id,
          "amount": -invoice.totalAmount, // Note the negative sign is here
          "currencyRate": 1
        }
      ],
      "allocatedOnDate": invoicePaymentDate
    }
  ],
}
```
</TabItem>

</Tabs>

> ### Create bank transactions

Author: ```suggestion
### Create bank feed transactions
```
